### ISSUE TITLE
Role-based access control on the front end.

### ISSUE THREAD

**Issue by AlexZhidkov** [AlexZhidkov/ProjectMarket#4]
Every user object in the database has a `role` property. At the moment possible values are "student", "business", "referrer" and "admin".
All new users should be assigned "student" role unless they came in through `./business/*` URL in which case they assigned "business" role.
We need to restrict access to URLs so students only can access `./student/*` and businesses only `./business/*` URLs.

Implement child routes for student and business and guard them with CanActivateChild 
https://angular.io/guide/router#canactivatechild-guarding-child-routes

**Comment by Estex** [comment]
I am not sure that implement lazy loading will be good idea, because to make it we will need to move student and business to separated 2 modules, and i see here few problems(even i know its good practice):
 
1) Time that need to be spend on change code structure 
2) I think that time for development in future will increase, as we little bit complicate structure

And as i understand right now we build prototype and more important to have it done as fast as possible, so i am not sure that its make sense to implement lazy loading. What do you think? 

Also i have 3rd problem - as i don`t have enough knowledge about app work, i am little bit afraid to change structure. But if you think its necessary now - i can do it

**Comment by AlexZhidkov** [comment]
@Estex Thanks for your considerations. I agree with you that lazy loading adds complexity. Can you estimate the extra work required to implement it? 
I want to do it because Student and Business users are going to be using entirely different app components, and their functionality doesn't intersect. We were thinking originally to build two separate apps for them.  And I want to do it now because the app is relatively small, and it will set us on the right path to develop further components. Lazy loading is made easier with the latest Angular, and I think we can justify doing it now.
If you don't feel comfortable doing it, please give me your estimate for this work before starting.

**Comment by AlexZhidkov** [comment]
@Estex I agree with you. I'll take lazy loading out into a separate issue. Thank you.

**Comment by AlexZhidkov** [comment]
@Estex I've added some ToDo comments to routing module. We need Referrer and Admin sections similar to Student and Business. HomeComponent should become Admin Dashboard. '**' path can be removed for now. Would you be able to make these changes?

**Comment by Estex** [comment]
@AlexZhidkov can you explain why we need to remove home path and why HomeComponent will be Admin Dashboard ? I am not sure that understand what i do :)
And i made pull request for  Referrer and Admin sections, without changing HomeComponent for now

**Comment by AlexZhidkov** [comment]
Students and businesses will have separate home pages. They should not crossover. They will have two totally separate app experience. Current home page was used for the testing purposes only.
